format/fit/fit.go

```diff
@@ -316,6 +324,30 @@ func fitDecodeDataMessage(d *decode.D, drc *dataRecordContext, lmfd localFieldDe
 
 	isDevDep := isDevMap[drc.localMessageType]
 
+	// pre read all integer fields and store them in the value map
+	// so that they can be referenced by eventual subfields
+	var val uint64
+	curPos := d.Pos()
+	for _, k := range keys {
+		fDef := lmfd[drc.localMessageType][uint64(k)]
+		switch fDef.Type {
+		case "enum", "uint8", "uint8z", "byte":
+			val = peekUintBitsEndian(d, 8)
+			valMap[fDef.Name] = valueType{value: val, typ: fDef.Format}
```

Comment:
But yes maybe something like `PeekUintBitsEndian` could be useful... wonder even if `peekUintBits` should actually respect current endian? or maybe those should be `d.Peek<type>` like `d.PeekU(24)`? could probably even generate code for those.

Reply:
Dammit =) I was looking at the d.U (and all the d.Ux, d.UxLE etc) functions in the lounge at the hotel in Brussels, and now they completely slipped my mind. Yes, d.U would work and look better. I'll fix it.

Reply:
😄 maybe the decode API is not the most "discoverable", have some vague idea how maybe it could be made nicer, maybe use generic and methods chaining more? but the current keep it "dumb" and generate-design it quite nice also :)

Reply:
And just as you merged, I realized that although there are no reference fields that are also array types, there might be some that would throw of the value gathering, so I need to fix that too. =) another PR incoming soon

Reply:
Ok, no worries!